@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1543,8 +1547,12 @@ export class TextElementsRenderer {
         tempScreenPosition.x = tempPoiScreenPosition.x = screenPosition.x;
         tempScreenPosition.y = tempPoiScreenPosition.y = screenPosition.y;
 
-        // Scale the text depending on the label's distance to the camera.
-        const textDistance = this.m_viewState.worldCenter.distanceTo(position);
+        // Scale the text depending on the label's distance to the camera "zero" plane.
+        const textDistance = pointToPlaneDistance(
+            position,
+            this.m_viewState.worldCenter,
+            this.m_cameraLookAt
+        );
```

Comment:
You will probably ask me why I did not used exactly the same function:
```
computeViewDistance(pointLabel, ...)
``` 
with pointLabel (TextElement) as input parameter, unfortunately quick research shows that it is not that simple and sometimes the labels are created and positioned with slightly different position then stored in their state. So I need the position from `addPointLabel` function. See the calls to `addPointLabel` it will probably explain everything.